Repository/ContainerRepositoryFactory.php

```diff
@@ -50,8 +50,15 @@ public function getRepository(EntityManagerInterface $entityManager, $entityName
             if ($this->container && $this->container->has($customRepositoryName)) {
                 $repository = $this->container->get($customRepositoryName);
 
-                if (! $repository instanceof EntityRepository) {
-                    throw new \RuntimeException(sprintf('The service "%s" must extend EntityRepository (or a base class, like ServiceEntityRepository).', $repositoryServiceId));
+                if (\interface_exists(ObjectRepository::class)) {
+                    if (!$repository instanceof ObjectRepository) {
+                        throw new \RuntimeException(sprintf('The service "%s" must implement ObjectRepository (or extend a base class, like ServiceEntityRepository).', $repositoryServiceId));
+                    }
+                } else {
+                    // Doctrine 2.4 compatibility layer
```

Comment:
ORM's `EntityRepository` has implemented the `ObjectRepository` interface since Doctrine 2.1. Thus, this compat layer isn't necessary.

Reply:
~~ I did it after seeing an inappropriate comment in Doctrine 2.4 codebase after [discussion here](https://github.com/doctrine/DoctrineBundle/issues/856). But you're absolutely right. I rollback that change.